src/main.c

```diff
@@ -433,6 +481,72 @@ void WriteInterfaces()
     free(InterfaceArray);
 }
 
+void ParseRasNdisWanPacketFragment(PEVENT_RECORD ev)
+{
+    // Parse the current RasNdisWan packet event for use elsewhere.
+    // NB: Here we only do per-packet parsing. For any event fields that only need to be
+    // parsed once and written into an INTERFACE, we do the parsing in AddInterface.
+    ULONG Err;
+    wchar_t buffer[256];
+    PROPERTY_DATA_DESCRIPTOR Desc;
+    ULONG ParamNameSize = 0;
+
+    // RoutingDomainID
+    Desc.PropertyName = (unsigned long long)(L"RoutingDomainID");
+    Desc.ArrayIndex = ULONG_MAX;
+    (void)TdhGetPropertySize(ev, 0, NULL, 1, &Desc, &ParamNameSize);
+    RasNdisWanPacketFragment.RoutingDomainID = malloc((ParamNameSize / sizeof(wchar_t)) + 1);
```

Comment:
Calling malloc on a per-packet basis is not allowed in etl2pcapng. This is the fastest converter on the planet and it will stay that way.

(p.s. you are currently leaking these allocations, but that's not a big deal because you're going to remove the mallocs!)

(p.s., a hint if you can't think of a good hardcoded buffer size: if you've already called malloc for a past packet, you can reuse that buffer. You only need to free that old buffer and alloc a new one if the current packet needs a bigger buffer.)